mobius-core/src/test/java/com/spotify/mobius/MobiusLoopBehaviorWithEventSources.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * -\-\-
+ * Mobius
+ * --
+ * Copyright (c) 2017-2018 Spotify AB
+ * --
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+package com.spotify.mobius;
+
+import static com.spotify.mobius.Next.next;
+import static com.spotify.mobius.Next.noChange;
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+
+import com.spotify.mobius.functions.Consumer;
+import com.spotify.mobius.runners.WorkRunners;
+import com.spotify.mobius.test.RecordingConsumer;
+import com.spotify.mobius.test.RecordingModelObserver;
+import com.spotify.mobius.testdomain.EventWithSafeEffect;
+import com.spotify.mobius.testdomain.TestEffect;
+import com.spotify.mobius.testdomain.TestEvent;
+import java.util.concurrent.Semaphore;
+import javax.annotation.Nonnull;
+import org.junit.Test;
+
+public class MobiusLoopBehaviorWithEventSources extends MobiusLoopTest {
+
+  @Test
+  public void invokesEventSourceOnlyOnModelUpdates() {
```

Comment:
@pettermahlen I added this test to demonstrate the desired behavior with event sources and renamed the other one to reflect a better description of what it's actually testing.